website/static/js/apiApplication.js

```diff
@@ -0,0 +1,433 @@
+            });
+        }.bind(this));
+        return request;
+    },
+    createApplication: function () {
+        var request = this.client.createOne(this.appData());
+        request.done(function (dataObj) {
+            this.appData(dataObj);
+            this.originalValues(dataObj.serialize());
+
+            this.changeMessage(language.apiOauth2Application.creationSuccess, 'text-success', 5000);
+            this.apiDetailUrl(dataObj.apiDetailUrl); // Toggle ViewModel --> act like a display view now.
+            historyjs.replaceState({}, '', dataObj.webDetailUrl);  // Update address bar to show new detail page
+        }.bind(this));
+
+        request.fail(function (xhr, status, error) {
+            $osf.growl('Error',
+                       language.apiOauth2Application.dataSendError,
+                       'danger');
+
+            Raven.captureMessage('Error registering new OAuth2 application', {
+                url: this.apiDetailUrl,
+                status: status,
+                error: error
+            });
+        }.bind(this));
+    },
+    submit: function () {
+        // Validate and dispatch form to correct handler based on view type
+        if (!this.appData().isValid()) {
+            // Turn on display of validation messages
+            this.showMessages(true);
+        } else {
+            this.showMessages(false);
+            if (this.isCreateView()) {
+                this.createApplication();
+            } else {
+                this.updateApplication();
+            }
+        }
+    },
+    deleteApplication: function () {
+        var appData = this.appData();
+        bootbox.confirm({
+            title: 'Deactivate application?',
+            message: language.apiOauth2Application.deactivateConfirm,
+            callback: function (confirmed) {
+                if (confirmed) {
+                    var request = this.client.deleteOne(appData );
+                    request.done(function () {
+                        this.allowExit(true);
+                        // Don't let user go back to a deleted application page
+                        historyjs.replaceState({}, '', this.webListUrl);
+                        this.visitList();
+                    }.bind(this));
+                    request.fail(function () {
+                            $osf.growl('Error',
+                                       language.apiOauth2Application.deactivateError,
+                                       'danger');
+                    }.bind(this));
+                }
+            }.bind(this),
+            buttons:{
+                confirm:{
+                    label:'Deactivate',
+                    className:'btn-danger'
+                }
+            }
+        });
+    },
+    visitList: function () {
+        window.location = this.webListUrl;
+    },
+    cancelChange: function () {
+        if (!this.dirty()) {
+            this.visitList();
+        } else {
+            bootbox.confirm({
```

Comment:
The style guide suggests:

> USE confirmation when:
> - The action will override an existing content AND the content can't be easily reverted.(e.g. Deleting files, wiki pages)

> DON'T USE confirmation when:
> - The action can easily be reverted (e.g instance settings changes that can be changed back)

This falls under the category of actions that can be reverted. Confirmation modals should be used sparingly, reserved mainly for destructive, irreversible actions. Overusing them not only interrupts user flow, but it also diminishes the protective value they're supposed to give (because the user becomes accustomed to ignoring them).